constraints/src/main/java/gov/nasa/jpl/aerie/constraints/tree/ForEachActivitySpans.java

```diff
@@ -61,8 +62,7 @@ public String prettyPrint(final String prefix) {
 
   @Override
   public boolean equals(Object obj) {
-    if (!(obj instanceof ForEachActivitySpans)) return false;
-    final var o = (ForEachActivitySpans)obj;
+    if (!(obj instanceof final ForEachActivitySpans o)) return false;
```

Comment:
Could you clarify what is happening in the `if` statement here?

Reply:
This is a non-behavioral change. Its syntactic sugar for context-aware downcasting.

```java
if (obj instanceof MySubclass) {
    final MySubclass obj2 = (MySubclass) obj;
    // do things
}

// is the same as

if (obj instanceof final MySubclass obj2) {
    // do things
}
```

Its even funkier if you check that it is *not* an instanceof the subclass, *AND* the if statement always breaks/returns/throws to disrupt control flow:

```java
if (!(obj instanceof MySubclass)) {
    // do things
    return; // necessary
}
// if this point is reached you know obj is an instance of MySubclass
final MySubclass obj2 = (MySubclass) obj;
// do other things

// is the same as

if (!(obj instance of final MySubclass obj2)) {
    // do things
    return; // necessary
}
// do more things
```

Reply:
Thanks for the explanation on context-aware downcasting!